### DISCUSSION TITLE
Extending `asdf` manager with community plugins used in Android/Kotlin projects

### DISCUSSION THREAD

**Discussion by ubiratansoares** [renovatebot/renovate#21609] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Hi team, hope you all are doing great! 👋🏻 

I'm starting this discussion since I see an opportunity to leverage `renovate` to drive automated updates for CLI tooling managed by `asdf` in the context of Android and/or Kotlin projects. I'm referring to tools like:

- [pinterest/ktlint](https://github.com/pinterest/ktlint)
- [detekt/detekt](https://github.com/detekt/detekt)
- [mobile-dev-inc/maestro](https://github.com/mobile-dev-inc/maestro)

which have asdf-plugins around, but we don't have in the [official upgradable tooling](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/asdf/upgradeable-tooling.ts) for `asdf`

I already have a fork of `renovate` and a local version with those `asdf` plugins added, but after having a chat with @secustor (we work in the same company 😄), I decided to raise a discussion rather than an issue, since I'd be happy to drive this initiative myself. It would be my very my first contribution to this project ❤️  

As a first step, my proposition would be starting with the 3 tools mentioned above, since they are straightforward and quite relevant in the Android ecosystem; also, they use `github-releases` as data sources and don't require fancy extraction rules for versions.

After that, I'd love to extend `asdf` manager with other tools we use for Android/Kotlin like [facebook/ktfmt](https://github.com/facebook/ktfmt), but those require a `maven` data source, so I see that as a second step.

Happy to raise a PR and provide a reproducer repo, just let me know 🙂

**Comment by secustor** [comment]
As discussed, you can create a PR for all 3 of these tools together.

**Reply by ubiratansoares** [reply]
Done!  🚀 

https://github.com/renovatebot/renovate/pull/21999
